catalog/CatalogRelation.hpp

```diff
@@ -190,6 +191,22 @@ class CatalogRelation : public CatalogRelationSchema {
 #endif  // QUICKSTEP_HAVE_LIBNUMA
 
   /**
+   * @brief Get the number of tuples in the catalog relation.
+   *
+   * @return The number of tuples in the catalog relation
+   **/
+  const std::size_t getNumTuples() const {
+    return num_tuples_;
+  }
+
+  /**
+   * @brief Set the number of tuples for catalog relation.
+   *
+   * @param num_tuples The number of tuples in the catalog relation.
+   **/
+  void setNumTuples(std::size_t num_tuples) const;
```

Comment:
Remove `const` since this method is actually updating the state of `CatalogRelation`.